packages/amplify-category-geo/src/provider-controllers/import.ts

```diff
@@ -113,4 +119,26 @@ const bulkUploadGeofence = async (params: GeofenceCollectionParams, region: stri
     throw new Error(`The uploaded geofences should have at most ${MAX_ENTRIES_PER_BATCH} items per batch.`);
   }
   return await service.batchPutGeofence(params).promise();
-}
\ No newline at end of file
+};
+
+/**
+ * Scan the custom properties as candidate options.
+ * @param geoJSONObj Geo features collection
+ * @returns Array of candidate custom property names which exist in all features
+ */
+const scanCandidateCustomProperties = (geoJSONObj: FeatureCollection): string[] => {
+  const { features } = geoJSONObj;
+  if (features && features.length > 0 && features[0].properties) {
+    const candidateProperties: string[] = Object.keys(features[0].properties).filter(prop => {
+      for (let i = 1; i < features.length; i += 1) {
+        // Remove the property that does not exist in all features
+        if (!features[i].properties[prop]) {
```

Comment:
Might need to change here as well.
```suggestion
        if (!features[i]?.properties?.prop) {
```